hoomd/md/pair.py

```diff
@@ -1866,50 +1866,24 @@ class ai_pair(pair):
     #  - self.required_coeffs (a list of the coeff names the derived class needs)
     #  - self.process_coeffs() (a method that takes in the coeffs and spits out a param struct to use in
     #       self.cpp_force.set_params())
-    def __init__(self, r_cut, nlist, name=None):
-        # initialize the base class
-        force._force.__init__(self, name);
-
-        self.global_r_cut = r_cut;
-
-        # setup the coefficient matrix
-        self.pair_coeff = coeff();
-        self.pair_coeff.set_default_coeff('r_cut', self.global_r_cut);
 
-        # setup the neighbor list
-        self.nlist = nlist
-        self.nlist.subscribe(lambda:self.get_rcut())
-        self.nlist.update_rcut()
+    def __init__(self, nlist, r_cut=None, r_on=0., mode='none'):
+        self._nlist = validate_nlist(nlist)
+        r_cut = float if r_cut is None else float(r_cut)
+        r_cut = TypeParameter('r_cut', 'particle_types',
+                              TypeParameterDict(r_cut, len_keys=2)
+                             )
+        r_on = TypeParameter('r_on', 'particle_types',
+                             TypeParameterDict(float(r_on), len_keys=2)
+                            )
+        self._extend_typeparam([r_cut, r_on])
+        self._param_dict.update(
+            ParameterDict(mode=OnlyFrom(['none', 'shifted', 'xplor']),
+                          explicit_defaults=dict(mode=mode))
+            )
```

Comment:
This is identical to `_Pair.__init__`. No need to have this code.

Reply:
This is also where the error in initializing you were getting comes from. The argument should be `_defaults` now not `explicit_defaults`.